flict/flictlib/project/reader.py

```diff
@@ -170,6 +183,22 @@ def _read_spdx_2_2(self, only_packages=None):
                     _pkg_name = _pkg['id']
                     if dep_package_name == _pkg_name:
                         packages[top_package]['dependencies'].append({_pkg_name: _pkg})
+            else:
+                dep_id = dep['spdxElementId']
```

Comment:
Are you sure about this? I would have expected that you only use the remainder after the `:` for looking up the package.
Also I'm missing a test case for both scenarios (package found and package not found).

Reply:
There should be no ":" in dep['spdxElementId'] in this branch so we should be able to use as is. Or?

Reply:
Should a missing package (one package refers to another package which is not defined) be raising an error instead? 

My vote: yes

Reply:
yes. please

Reply:
Keep warning for now.
Adding an issue later.